zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/util/ZeebeAssertHelper.java

```diff
@@ -330,4 +331,24 @@ public static void assertVariableDocumentUpdated(
     assertThat(record).isNotNull();
     eventConsumer.accept(record.getValue());
   }
+
+  public static void assertUserCreated(final String username) {
+    assertUserCreated(username, u -> {});
+  }
+
+  public static void assertUserCreated(
+      final String username, final Consumer<UserRecordValue> consumer) {
+    final UserRecordValue user =
+        RecordingExporter.userRecords()
+            .filter(
+                userRecordValueRecord ->
+                    userRecordValueRecord.getValue().getUsername().equals(username))
+            .limit(1)
+            .map(Record::getValue)
+            .toList()
+            .getLast();
```

Comment:
❌ This filter should be on the `UserRecordStream`. That way it's easier to use and reuse. You can just say `.withUsername(username)` here.

I noticed on the `UserRecordStream` we already have this filter, it's just named `withUsernameKey`. Let's rename that to `withUsername`. The key part is not relevant and would only cause confusion as it's not a key.


You also don't filter for intent. This means that you are searching for all user record, not just the created ones. It works because you pick the last record, which is the created record in this case. But it's a bit fault-sensitive when we make changes in the future.


In the end I'd expect it to look like this:

```suggestion
    final var user =
        RecordingExporter.userRecords()
            .withIntent(UserIntent.CREATED)
            .withUsername(username)
            .getFirst()
            .getValue();
```